nixos/modules/services/networking/searx.nix

```diff
@@ -47,16 +47,16 @@ in
 
   config = mkIf config.services.searx.enable {
 
-    users.users.searx =
-      { uid = config.ids.uids.searx;
-        description = "Searx user";
-        createHome = true;
-        home = "/var/lib/searx";
-      };
+   users.users.searx = {
+      uid = config.ids.uids.searx;
+      description = "Searx user";
+      createHome = true;
+      home = "/var/lib/searx";
```

Comment:
You should use the systemd `StateDirectory` mechanism, instead.
Basically, remove the two lines here and add a `serviceConfig.StateDirectory = "searx";`, this will create a directory with the right permissions.

Reply:
Done, thanks for the tip.